lib/archive/diff.go

```diff
@@ -162,35 +195,40 @@ func createHeader(op trace.Operation, dir string, change docker.Change, spec *Fi
 		whiteOutDir := filepath.Dir(change.Path)
 		whiteOutBase := filepath.Base(change.Path)
 		whiteOut := filepath.Join(whiteOutDir, docker.WhiteoutPrefix+whiteOutBase)
+
+		// FIXME: consult @jzt about how strip should work here
 		hdr = &tar.Header{
-			Name:       filepath.Join(spec.RebasePath, whiteOut),
 			ModTime:    timestamp,
 			AccessTime: timestamp,
 			ChangeTime: timestamp,
 		}
+
+		whiteOut = strings.TrimPrefix(whiteOut, "/")
+		strippedName := strings.TrimPrefix(whiteOut, spec.StripPath)
+		hdr.Name = filepath.Join(spec.RebasePath, strippedName)
 	default:
 		fi, err := os.Lstat(filepath.Join(dir, change.Path))
 		if err != nil {
 			op.Errorf("Error getting file info: %s", err.Error())
 			return nil, err
 		}
 
-		hdr, err = tar.FileInfoHeader(fi, change.Path)
+		link, err := os.Readlink(filepath.Join(dir, change.Path))
+		if err != nil {
+			link = change.Path
+		}
+		hdr, err = tar.FileInfoHeader(fi, link)
 		if err != nil {
 			op.Errorf("Error getting file info header: %s", err.Error())
 			return nil, err
 		}
 
-		hdr.Name = filepath.Join(spec.RebasePath, change.Path)
-
-		if hdr.Typeflag == tar.TypeDir {
-			hdr.Name += "/"
-		}
+		change.Path = strings.TrimPrefix(change.Path, "/")
+		strippedName := strings.TrimPrefix(change.Path, spec.StripPath)
+		change.Path = strings.TrimPrefix(change.Path, "/")
+		hdr.Name = filepath.Join(spec.RebasePath, strippedName)
 	}
 
```

Comment:
I believe this still needs to be addressed, out of band with @jzt it is determined that this is a must.